test/run_tests_model_gen_and_load.py

```diff
@@ -0,0 +1,236 @@
+import argparse
+from genericpath import isdir, isfile
+from pathlib import Path
+import re
+import os
+import shutil
+import json
+from run_tests import convert_to_test_data, Color, NoColor
+import vowpalwabbit
+
+
+"""
+This script will use run_tests to capture all available tests to run
+Its goal is to generate models (--generate_models) for all tests that are eligible (see get_tests() for the tests that are excluded)
+by using the vw python interface and just passing in the command line of the runtest
+
+Most, if not all, runtests have the -d option specified so using the cli directly in the python interface will train a model using the file specified.
+
+After the models are generated we can run the same runtests that will load each model (--load_models) by using the same python interface,
+and that will result in the datafile being processed again
+
+The idea is that we can generate the models using one vw version and load the models in an older (or newer) vw version and check for forwards (backwards) model compatibility
+"""
+
+default_working_dir_name = ".vw_runtests_model_gen_working_dir"
+default_test_file = "core.vwtest.json"
+
+
+def create_test_dir(test_id, input_files, test_base_dir, test_ref_dir):
+    for f in input_files:
+        file_to_copy = None
+        search_paths = [test_ref_dir / f]
+
+        for search_path in search_paths:
+            if search_path.exists() and not search_path.is_dir():
+                file_to_copy = search_path
+                break
+
+        if file_to_copy is None:
+            raise ValueError(
+                f"Input file '{f}' couldn't be found for test {test_id}. Searched in '{test_ref_dir}'"
+            )
+
+        test_dest_file = test_base_dir / f
+        if file_to_copy == test_dest_file:
+            continue
+        test_dest_file.parent.mkdir(parents=True, exist_ok=True)
+        # We always want to replace this file in case it is the output of another test
+        if test_dest_file.exists():
+            test_dest_file.unlink()
+
+        shutil.copy(str(file_to_copy), str(test_dest_file))
+
+
+def generate_model(test_id, command, working_dir, color_enum=Color):
+    command = command + " --quiet "
+    print(f"{color_enum.LIGHT_CYAN}id: {test_id}, command: {command}{color_enum.ENDC}")
+    vw = vowpalwabbit.Workspace(command)
+
+    vw.save(f"{working_dir}/model_{test_id}.vw")
```

Comment:
construct the path using pathlib then convert it. Not sure if `vw.save` supports `Path` input or not
```suggestion
    vw.save(str(working_dir / f"model_{test_id}.vw"))
```

Reply:
it is being stringified before it is passed into vw.save already i.e. `f"{...}"`

Reply:
but sure